src/components/implementation/no_interface/cFE_booter/ostask.c

```diff
@@ -0,0 +1,815 @@
+
+	if (priority > 255 || priority < 1) { return OS_ERR_INVALID_PRIORITY; }
+
+	struct sl_thd *thd = sl_thd_alloc(osal_task_entry_wrapper, function_pointer);
+	assert(thd);
+	union sched_param_union sp = {.c = {.type = SCHEDP_PRIO, .value = priority}};
+	sl_thd_param_set(thd, sp.v);
+
+	struct cfe_task_info *task_info = &cfe_tasks[sl_thd_thdid(thd)];
+	strcpy(task_info->osal_task_prop.name, task_name);
+	task_info->osal_task_prop.creator    = OS_TaskGetId();
+	task_info->osal_task_prop.stack_size = stack_size;
+	task_info->osal_task_prop.priority   = priority;
+	task_info->osal_task_prop.OStask_id  = (uint32)sl_thd_thdid(thd);
+	task_info->delete_handler            = NULL;
+
+	*task_id = (uint32)sl_thd_thdid(thd);
+
+	return OS_SUCCESS;
+}
+
+int32
+OS_TaskDelete(uint32 task_id)
+{
+	struct sl_thd *thd = sl_thd_lkup(task_id);
+	if (!thd) { return OS_ERR_INVALID_ID; }
+
+	struct cfe_task_info *task_info = &cfe_tasks[task_id];
+
+	osal_task_entry delete_handler = task_info->delete_handler;
+	if (delete_handler) { delete_handler(); }
+
+	sl_thd_free(thd);
+
+	return OS_SUCCESS;
+}
+
+uint32
+OS_TaskGetId(void)
+{
+	thdid_t real_id           = sl_thdid();
+	thdid_t possible_override = id_overrides[real_id];
+	if (possible_override) return possible_override;
+	return real_id;
+}
+
+void
+OS_TaskExit(void)
+{
+	sl_thd_free(sl_thd_curr());
+	PANIC("Should be unreachable!");
+}
+
+int32
+OS_TaskInstallDeleteHandler(osal_task_entry function_pointer)
+{
+	if (OS_TaskGetId() == main_delegate_thread_id) { return OS_ERR_INVALID_ID; }
+
+	struct cfe_task_info *task_info = &cfe_tasks[sl_thd_thdid(sl_thd_curr())];
+	task_info->delete_handler = function_pointer;
+
+	return OS_SUCCESS;
+}
+
+int32
+OS_TaskDelay(uint32 millisecond)
+{
+	cycles_t wakeup = sl_now() + sl_usec2cyc(millisecond * 1000);
+	sl_thd_block_timeout(0, wakeup);
+	return OS_SUCCESS;
+}
+
+int32
+OS_TaskSetPriority(uint32 task_id, uint32 new_priority)
+{
+	if (new_priority > 255 || new_priority < 1) { return OS_ERR_INVALID_PRIORITY; }
+
+	struct sl_thd *thd = sl_thd_lkup(task_id);
+	if (!thd) { return OS_ERR_INVALID_ID; }
+
+	union sched_param_union sp = {.c = {.type = SCHEDP_PRIO, .value = new_priority}};
+	sl_thd_param_set(thd, sp.v);
+
+	return OS_SUCCESS;
+}
+
+int32
+OS_TaskRegister(void)
+{
+	if (OS_TaskGetId() == main_delegate_thread_id) { return OS_ERR_INVALID_ID; }
+
+	return OS_SUCCESS;
+}
+
+
+int32
+OS_TaskGetIdByName(uint32 *task_id, const char *task_name)
+{
+	if (!task_id || !task_name) return OS_INVALID_POINTER;
+
+	thdid_t i;
+	for (i = 1; i < SL_MAX_NUM_THDS; i++) {
+		struct sl_thd *thd = sl_thd_lkup(i);
+		if (!thd || thd->state == SL_THD_FREE) continue;
+		if (strcmp(cfe_tasks[i].osal_task_prop.name, task_name) == 0) {
+			*task_id = i;
+			return OS_SUCCESS;
+		}
+	}
+
+	return OS_ERR_NAME_NOT_FOUND;
+}
+
+int32
+OS_TaskGetInfo(uint32 task_id, OS_task_prop_t *task_prop)
+{
+	if (!task_prop) { return OS_INVALID_POINTER; }
+
+	struct sl_thd *thd = sl_thd_lkup(task_id);
```

Comment:
def at head of blocks

Reply:
OK, I'm going to stop listing these so as to not be to annoying.  You can fix this throughout easily.